clients/roscpp/src/libros/service_server_link.cpp

```diff
@@ -362,14 +363,34 @@ bool ServiceServerLink::call(const SerializedMessage& req, SerializedMessage& re
     processNextCall();
   }
 
+  bool interrupted = false;
   {
     boost::mutex::scoped_lock lock(info->finished_mutex_);
 
+    using namespace boost::chrono;
+    system_clock::time_point now = system_clock::now();
```

Comment:
IMHO this should use the `steady_clock` to not be affected by system time changes as in #1250 and #1014